examples/custom-databases/mongoose.js

```diff
@@ -0,0 +1,116 @@
+const Discord = require('discord.js'),
+    client = new Discord.Client(),
+    settings = {
+        prefix: 'g!',
+        token: 'Your Discord Bot Token',
+    };
+
+// Connect to database
+const mongoose = require('mongoose');
+mongoose.connect('mongodb://localhost/giveaways', { useFindAndModify: false });
+const db = mongoose.connection;
+
+// Check the connection
+db.on('error', console.error.bind(console, 'Connection error:'));
+db.once('open', () => {
+    console.log('Connected to MongoDB.');
+});
+
+// Create the schema for giveaways
+const giveawaySchema = new mongoose.Schema({
+    messageID: String,
+    channelID: String,
+    guildID: String,
+    startAt: Number,
+    endAt: Number,
+    ended: Boolean,
+    winnerCount: Number,
+    prize: String,
+    messages: {
+        giveaway: String,
+        giveawayEnded: String,
+        inviteToParticipate: String,
+        timeRemaining: String,
+        winMessage: String,
+        embedFooter: String,
+        noWinner: String,
+        winners: String,
+        endedAt: String,
+        hostedBy: String,
+        units: {
+            seconds: String,
+            minutes: String,
+            hours: String,
+            days: String,
+            pluralS: Boolean,
+        },
+    },
+    hostedBy: String,
+    winnerIDs: [],
+    reaction: String,
+    botsCanWin: Boolean,
+    embedColor: String,
+    embedColorEnd: String,
+    exemptPermissions: [],
+    extraData: {}
+});
+
+// Create the model
+const giveawayModel = mongoose.model('giveaways', giveawaySchema);
+
+const { GiveawaysManager } = require('discord-giveaways');
+class GiveawayManagerWithMongoose extends GiveawaysManager {
+    // This function is called when the manager needs to get all the giveaways stored in the database.
+    async getAllGiveaways() {
+        // Get all the giveaways in the database. We can use the messageID field to fetch all documents by passing an empty condition.
+        return await giveawayModel.find({}, 'messageID');
```

Comment:
```suggestion
        return await giveawayModel.find({});
```
If you have the 'messageID' then it will only return the messageIDs like you see below but it should return the whole objects. That's why Ethan/V1rtuaI'S giveaways didn't update after a restart because the bot only returned messageIDs with `getAllGiveaways`
![image](https://user-images.githubusercontent.com/63612668/103557117-ff577900-4eb2-11eb-8fb9-36736336818c.png)

Did you really test what happens after a restart? Did it really update when you used 'messageID'? Because i saw Ethan's code and tested it with him and it did clearly not until we removed the 'messageID' and then it finally returned the whole giveaway objects.